src/geometry/ArborX_GeometryTraits.hpp

```diff
@@ -50,13 +53,23 @@ struct tag
   using type = not_specialized;
 };
 
+template <typename Geometry>
+struct coordinate_type
+{
+  using type = not_specialized;
```

Comment:
Why did we introduce this new trait?  Why is it not something we deduce from indexing into a point coordinate?

Reply:
We could do that for point, or box, or anything that has access operator. For sphere or triangle, it is harder to do. If i pass a general Geometry, and want to know the type it's based on, I'm not sure there is an easy way.

Reply:
For sphere, with our current interface, you have access to the centroid and radius.
Similarly triangles are composed of triangles.  I understand that some changes might be needed to deduce what is called `coordinate_type` in this PR, I just don't think introducing a new traits was the right thing to do.

Reply:
I was not sure, so i decided to follow [Boost Geometry](https://www.boost.org/doc/libs/1_82_0/libs/geometry/doc/html/geometry/reference/core/coordinate_type.html).

If you think this is a wrong way to go, and have a better way, we should discuss.